Bottom: e64f4d4550097124d1308243aaf6aefe72febded
Top:    518dce469fbaa9c1ae345cdf4610d940f0283c41
Author: Ian Kent <ikent@redhat.com>
Date:   2013-04-04 14:57:40 +0800

Refresh of autofs-5.0.7-syncronize-handle_mounts-shutdown.patch

---

diff --git a/daemon/automount.c b/daemon/automount.c
index a016903..476ff14 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -1354,7 +1354,8 @@ static void *statemachine(void *arg)
 					return NULL;
 				}
 			} else {
-				if (master_done(master_list)) {
+				master_done(master_list);
+				if (list_empty(&master_list->mounts)) {
 					master_mutex_unlock();
 					return NULL;
 				}
